src/components/Menu/DrawerMenu.module.css

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2023 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+.bg {
+  position: fixed;
+  z-index: 100;
```

Comment:
Good point, I copy/pasted those Z-index values from Element Call without considering the issues they might create. Unfortunately, since Vaul relies on a separate div element for the backdrop, rather than the `::backdrop` pseudo-element, I don't think `<dialog>` would fully solve the problem for us. But from what I can tell, everything still works the same if I leave `z-index` set to its default value of `auto`? So I think I'll go with that solution.